test/gtest/common/mem_buffer.cc

```diff
@@ -39,6 +39,23 @@ std::vector<ucs_memory_type_t> mem_buffer::supported_mem_types()
     return vec;
 }
 
+std::vector<std::pair<ucs_memory_type_t, ucs_memory_type_t> >
+mem_buffer::supported_mem_type_pairs()
+{
+    std::vector<ucs_memory_type_t> mem_types = mem_buffer::supported_mem_types();
+    std::vector<std::pair<ucs_memory_type_t, ucs_memory_type_t> > pairs;
+
+    for (std::vector<ucs_memory_type_t>::const_iterator i =
```

Comment:
this function should not be in mem_buffer, have some other helper to construct cartesian multiplication